### PR TITLE
feature: add user flag implement in pouch create and exec

### PR DESCRIPTION
**Pull request by Ace-Tang**
add user implement like format uid[user]:gid[group]
support command pouch exec -u and pouch create -u

Signed-off-by: Ace-Tang <aceapril@126.com>

### Ⅰ. Describe what this PR did

add user implement like format uid[user]:gid[group]
support command pouch exec -u and pouch create -u

### CODE REVIEW

**Comment by Letty5411** [comment]
@Ace-Tang Could you also add more verifications in test `TestRunWithUser` to verify this PR works.

**Comment by Ace-Tang** [comment]
@Letty5411 , yes , I can add more case in `TestStartWithUser`

File: pkg/user/user.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		if len(v) <= i {
+			break
+		}
+
+		switch p := v[i].(type) {
+		case *string:
+			*p = s
+		case *int:
+			*p, _ = strconv.Atoi(s)
+		case *[]string:
```
Comment (allencloud) [inline pkg/user/user.go]: I suggest that we could add test case in the unit test for this case.

File: pkg/user/user.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+		}
+		if ok {
+			return uint32(idInt), nil
+		}
+	}
+
+	return 0, fmt.Errorf("failed to find id or name %s in %s", str, file)
+}
+
+// isUnknowUser determines if id can be accepted as a unknow user. this kind of user id should >= 1000
```
Comment (allencloud) [inline pkg/user/user.go]: s/unknow/unknown ?

File: pkg/user/user.go (hunk 1 above)
Comment (Ace-Tang) [reply pkg/user/user.go]: already get `TestParseString` ut in test file

File: pkg/user/user.go (hunk 2 above)
Comment (Ace-Tang) [reply pkg/user/user.go]: yes, updated.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**